federation/executor.go

```diff
@@ -324,5 +337,6 @@ func (e *Executor) Execute(ctx context.Context, query *graphql.Query) (interface
 	if err != nil {
 		return nil, err
 	}
+	deleteKey(r, federationField)
```

Comment:
I moved delete key back to the end since it would minimize the number of times we would have to parse through all the entries in the query to delete the federation field. 

Before this PR it was nested in the execute function, but it wasn't recursively looking through the entire query. The union tests have a use case where the federtionField is in a nested field, so we would need to change the implementation which makes deleting the federation field in the execute function slightly more expensive